pkg/agent/cniserver/ipam/hostlocal/gc.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hostlocal
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/spf13/afero"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+)
+
+const dataDir = "/var/lib/cni/networks"
+
+func networkDir(network string) string {
+	return filepath.Join(dataDir, network)
+}
+
+// This is a hacky approach as we access the internals of the host-local plugin,
+// instead of using the CNI interface. However, crafting a CNI DEL request from
+// scratch would also be hacky.
+func GarbageCollectContainerIPs(network string, desiredIPs sets.Set[string]) error {
+	dir := networkDir(network)
+
+	info, err := os.Stat(dir)
+	if os.IsNotExist(err) {
+		klog.V(2).InfoS("Host-local IPAM data directory does not exist, nothing to do", "dir", dir)
+		return nil
+	}
+	if !info.IsDir() {
+		return fmt.Errorf("path '%s' is not a directory: %w", dir, err)
+	}
+
+	lk, err := NewFileLock(dataDir)
```

Comment:
It seems https://github.com/containernetworking/plugins/blob/93a1b3d0e71b3a4b5c463de357876fe8d383012f/plugins/ipam/host-local/backend/disk/backend.go#L53 uses `dir` instead of its parent dir `dataDir`?

Reply:
That's a pretty bad typo, fixed